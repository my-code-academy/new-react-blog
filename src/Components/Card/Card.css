  .date {
    font-size: 10px;
  }
  
  .title {
    font-size: 22px;
  }
  
  .abstract {
    font-size: 20px;
    color: #606060;
  }

  .card {
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    padding: 0px;
    text-align: center;
    /* background-color: #f1f1f1; */
    background-color: rgb(255, 255, 255);
    /* margin-left: 5%;
    margin-right: 5%;  */
    margin-top: 2%;
    width: 350px;
    padding: 5PX;
  }
  
  .container {
    text-align: left;
    font-family: Arial, Helvetica, sans-serif;
  }
  
  .clap {
    width: 10%;
    vertical-align: middle;
    margin-right: 5px;
  }
  
  .heart {
    width: 10%; 
    float: right;
  }
  
  .image {
    width: 100%;
    height: 50%;
  }
  
  #readTime {
    float: right;
  }
  /*
  #readTime {
    float: left;
  } */

  .dateTime {
    margin-top: 2%;
    display: flex;
    justify-content: space-between;
  }
