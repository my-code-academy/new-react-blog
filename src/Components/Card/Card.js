import React, { Component } from 'react';
import './Card.css';
import BlackHeartImage from '../../Icons/heart-black.svg'
import RedHeartImage from '../../Icons/heart-red.svg';
import ClappingImage from '../../Icons/clapping.svg';


class Card extends Component {
    state = {
        open: this.props.liked,
        clapsCount: this.props.claps
    }

    toggleImage = () => {
        this.setState(state => ({ open: !state.open }))
    }
    

    addClap = () => {
        this.setState(state => ({ clapsCount: state.clapsCount+1 }))
    }
    getImageName = () => this.state.open ? BlackHeartImage : RedHeartImage;
    render() {
        return (
            <div class="card">
                <img src={this.props.imageSrc} alt="abstract" className='image' />
                <div className="container">
                    <div className="dateTime">
                        <span className="date">{this.props.date}</span>
                        <span className='date' id="readTime">{this.props.readingTime}</span>
                    </div>
                    <p className='title'>{this.props.title}</p>
                    <p className='abstract'>{this.props.description}</p>
                    <hr />
                    <div className="responses">
                        <img onClick={this.addClap} alt="clap" src={ClappingImage} className='clap' />
                        <span>{this.state.clapsCount}</span>
                        <img alt="" className="heart" src={this.getImageName()} onClick={this.toggleImage}/>
                    </div>
                </div>
            </div>
        );
    }
}

export default Card;
