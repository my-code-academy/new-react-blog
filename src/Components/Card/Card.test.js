import React from 'react';
import renderer from 'react-test-renderer';
import Card from './Card';
import {shallow} from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Card', () => {
  it("should render a card", () => {
    const tree = renderer.create(<Card/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should increment clap count and toggle like", () => {
    const card = {
      "date": "2nd Januray, 2018",
      "readingTime": "2 mins",
      "title": "The future of abstract art and the culture ...",
      "description": "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
      "claps": 10,
      "liked": false,
      "image": "abstract.png"
    };

    const wrapper = shallow(<Card date = {card.date}
      readingTime = {card.readingTime}
      title = {card.title}
      description = {card.description}
      claps = {card.claps}
      liked = {card.liked}
      imageSrc = {`./Images/${card.image}`}  />);
    //console.log(wrapper.instance().renderRows());
    wrapper.instance().addClap();
    expect(wrapper.instance().state.clapsCount).toEqual(card.claps+1);
    wrapper.instance().toggleImage();
    expect(wrapper.instance().state.open).toEqual(!card.liked);
  });
});
