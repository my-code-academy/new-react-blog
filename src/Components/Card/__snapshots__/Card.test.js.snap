// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render a ard 1`] = `
<div
  class="card"
>
  <img
    alt="abstract"
    className="image"
  />
  <div
    className="container"
  >
    <div
      className="dateTime"
    >
      <span
        className="date"
      />
      <span
        className="date"
        id="readTime"
      />
    </div>
    <p
      className="title"
    />
    <p
      className="abstract"
    />
    <hr />
    <div
      className="responses"
    >
      <img
        alt="clap"
        className="clap"
        onClick={[Function]}
        src="clapping.svg"
      />
      <span />
      <img
        alt=""
        className="heart"
        onClick={[Function]}
        src="heart-red.svg"
      />
    </div>
  </div>
</div>
`;

exports[`Card should render a card 1`] = `
<div
  class="card"
>
  <img
    alt="abstract"
    className="image"
  />
  <div
    className="container"
  >
    <div
      className="dateTime"
    >
      <span
        className="date"
      />
      <span
        className="date"
        id="readTime"
      />
    </div>
    <p
      className="title"
    />
    <p
      className="abstract"
    />
    <hr />
    <div
      className="responses"
    >
      <img
        alt="clap"
        className="clap"
        onClick={[Function]}
        src="clapping.svg"
      />
      <span />
      <img
        alt=""
        className="heart"
        onClick={[Function]}
        src="heart-red.svg"
      />
    </div>
  </div>
</div>
`;
